crates/mcp-server/src/lib.rs

```diff
@@ -236,3 +236,28 @@ where
         Ok(())
     }
 }
+
+// Define a specific service implementation that we need for any
+// Any router implements this
+pub trait BoundedService:
```

Comment:
is there a more obvious name? (I assume this is fine, but not sure what it does without looking further?)